### ISSUE TITLE
[3.7] `ionic serve` stops after "build dev started" when `ionic serve` is already running for a different project

### ISSUE THREAD

**Issue by janpio** [ionic-team/ionic-cli#2619]
**Description:**

When I try to run `ionic serve` for a newly created project while I already have `ionic serve` running for another one, it fails:

```
PS C:\Users\Jan\Documents\stylingTest> ionic serve
[INFO] Starting app-scripts server: --port 8100 --p 8100 --livereload-port 35729 --r 35729 --address localhost - Ctrl+C
       to cancel
[13:23:19]  watch started ...
[13:23:19]  build dev started ...
PS C:\Users\Jan\Documents\stylingTest>
```

**Steps to Reproduce:**

Have 2 projects, start `ionic serve` in one, go to other folder, run `ionic serve`

**My `ionic info`:**

cli packages: (C:\Users\Jan\Documents\stylingTest\node_modules)

    @ionic/cli-plugin-ionic-angular : 1.4.1
    @ionic/cli-utils                : 1.7.0
    ionic (Ionic CLI)               : 3.7.0

local packages:

    @ionic/app-scripts : 2.1.3
    Ionic Framework    : ionic-angular 3.6.0

System:

    Android SDK Tools : 26.0.2
    Node              : v8.1.3
    OS                : Windows 10
    npm               : 5.1.0

**Comment by janpio** [comment]
Output of `--verbose`:

```
PS C:\Users\Jan\Documents\stylingTest> ionic serve --verbose
[DEBUG] CLI flags: { interactive: true, confirm: false }
[DEBUG] { cwd: 'C:\\Users\\Jan\\Documents\\stylingTest', local: true, binPath:
        'C:\\nvm\\v8.1.3\\node_modules\\ionic\\bin\\ionic', libPath:
        'C:\\Users\\Jan\\Documents\\stylingTest\\node_modules\\ionic\\dist\\index.js' }
[DEBUG] Loading local plugin @ionic/cli-plugin-proxy
[DEBUG] Throwing PLUGIN_NOT_INSTALLED for local @ionic/cli-plugin-proxy
[DEBUG] Loading local plugin @ionic/cli-plugin-ionic-angular
[DEBUG] Daemon found (pid: 13956)
[DEBUG] Getting plugin info for ionic
[DEBUG] Checking for latest plugin version of ionic@latest.
[DEBUG] Getting plugin info for @ionic/cli-plugin-ionic-angular
[DEBUG] Checking for latest plugin version of @ionic/cli-plugin-ionic-angular@latest.
[INFO] Starting app-scripts server: --log-level debug --port 8100 --p 8100 --livereload-port 35729 --r 35729 --address
       localhost - Ctrl+C to cancel
[13:26:23]  watch started ...
[13:26:23]  build dev started ...
PS C:\Users\Jan\Documents\stylingTest>
```

**Comment by janpio** [comment]
Running `npm run ionic:serve` directly works as expected.
Running `ionic serve --address=0.0.0.0` also work as expected:
```
PS C:\Users\Jan\Documents\stylingTest> ionic serve --address=0.0.0.0
[WARN] Multiple network interfaces detected!
       You will be prompted to select an external-facing IP for the livereload server that your device or emulator ha
       access to.
       You may also use the --address option to skip this prompt.

? Please select which IP to use: 192.168.10.1
[INFO] Starting app-scripts server: --address 0.0.0.0 --port 8100 --p 8100 --livereload-port 35729 --r 35729 - Ctrl+C
       cancel
[13:32:31]  watch started ...
[13:32:31]  build dev started ...
[13:32:31]  clean started ...
[13:32:31]  clean finished in 7 ms
[13:32:31]  copy started ...
[13:32:31]  transpile started ...
[13:32:35]  transpile finished in 4.06 s
[13:32:35]  preprocess started ...
[13:32:35]  deeplinks started ...
[13:32:35]  deeplinks finished in 14 ms
[13:32:35]  preprocess finished in 18 ms
[13:32:35]  webpack started ...
[13:32:36]  copy finished in 4.32 s
[13:32:47]  webpack finished in 11.32 s
[13:32:47]  sass started ...
[13:32:48]  sass finished in 1.45 s
[13:32:48]  postprocess started ...
[13:32:48]  postprocess finished in 9 ms
[13:32:48]  lint started ...
[13:32:48]  build dev finished in 16.96 s
[13:32:48]  watch ready in 17.11 s
[13:32:48]  dev server running: http://localhost:8100/

[INFO] Development server running
       Local: http://localhost:8100
       External: http://192.168.10.1:8100
```

**Comment by janpio** [comment]
Ok, these two command do something, but were rendering the wrong app - which I found running an `ionic serve` in another console window.

**Comment by dimitri320** [comment]
+1

**Comment by ValentinKlinghammer** [comment]
Interesting, I had the same problem. `ionic serve` simply ended.

However, I discovered that I had another `ionic serve` task running in a different project in another terminal window. So, at least for me, Ionic CLI 3.7 seems to fail if some of the ports are already bound, without an error message.

**Comment by dwieeb** [comment]
Hi all, please try with the new release CLI 3.8. I believe I've fixed the issue 👍 

Sorry about that!

**Comment by pavillet** [comment]
Had the same issue with CLI 3.8.1

I'm not sure this is the exact same source but for anyone looking into this : 

- Ionic Serve was stopping right after `[16:05:04]  copy finished in 2.96 s `
- No error, just stopping 
- I used `npm run ionic:serve` to see if there would be more info. There was in the form of this error 
` node.add(nextLine.substr(0, mapping.generatedColumn));
                         ^
TypeError: Cannot read property 'substr' of undefined`

- Leading me to this issue : https://github.com/ionic-team/ionic-app-scripts/issues/593

TLDR : In the end 2 components in the same files ends up throwing an error => split each component in one file

**Comment by gregor-srdic** [comment]
Omg, thanks, never would have thought of that! I also have two components in the same file for the first time ever :P

**Comment by idanb11** [comment]
I still cannot use ionic directly to serve or build my project:
(I don't have 2 components in the same file.)
```
$ ionic serve --lab
Starting app-scripts server: --address 0.0.0.0 --port 8100 --livereload-port 35729 --dev-logger-port 53703 --nobrowser --lab - Ctrl+C to cancel
[17:05:00]  watch started ...
[17:05:00]  build dev started ...

... and then Ionic just exits with no message or error
```

only when I ran this `$ npm run ionic:serve --lab` it works as expected...

No changes where made in the packages version or anything like that to causes this problem...
It just appeared out of nowhere...

**Comment by eai301** [comment]
same problem here 

npm -v
4.6.1

**Comment by chainarp** [comment]
Same Problem, My project was just created (ionic start MY_PROJECT blank)  no any touch witch it 

Starting app-scripts server: --address 0.0.0.0 --port 8100 --livereload-port
35729 --dev-logger-port 53703 --nobrowser - Ctrl+C to cancel
[22:18:13]  watch started ...
[22:18:13]  build dev started ...


cli packages: (C:\Users\Chainaris\AppData\Roaming\npm\node_modules)

    @ionic/cli-utils  : 1.19.0
    ionic (Ionic CLI) : 3.19.0

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.1.5
    Cordova Platforms  : none
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v6.11.2
    npm  : 3.10.10
    OS   : Windows 10

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro

**Comment by coloz** [comment]
same.
`PS C:\Users\coloz\Desktop\image2bitmap> npm run ionic:serve

> image2bitmap@0.0.1 ionic:serve C:\Users\coloz\Desktop\image2bitmap
> ionic-app-scripts serve

[15:03:19]  ionic-app-scripts 3.1.6
[15:03:19]  watch started ...
[15:03:19]  build dev started ...
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::53703
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at Server.setupListenHandle [as _listen2] (net.js:1351:14)
    at listenInCluster (net.js:1392:12)
    at Server.listen (net.js:1476:7)
    at new WebSocketServer (C:\Users\coloz\Desktop\image2bitmap\node_modules\ws\lib\WebSocketServer.js:77:20)
    at Object.createNotificationServer (C:\Users\coloz\Desktop\image2bitmap\node_modules\@ionic\app-scripts\dist\dev-server\notification-server.js:66:15)
    at C:\Users\coloz\Desktop\image2bitmap\node_modules\@ionic\app-scripts\dist\serve.js:47:31
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! image2bitmap@0.0.1 ionic:serve: `ionic-app-scripts serve`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the image2bitmap@0.0.1 ionic:serve script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\coloz\AppData\Roaming\npm-cache\_logs\2018-01-01T07_03_19_313Z-debug.log`

**Comment by Wolf00Bomber** [comment]
Same Issue with Ionic cli 3.19.1. 
When can we expect a fix for the same, Let us know.
Also this issue is not a issue ! in Other computers with fairly the same Env.

>Ionic info

@ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : not installed

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : none
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v8.9.4
    npm  : 5.6.0
    OS   : Windows 7

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro

**Comment by dwieeb** [comment]
The original issue has long been fixed and none of the recent comments have the same issue, so I'm locking this issue for now.

If you're seeing a problem with `ionic serve`, please create a new issue and fill out the issue template. I'll be happy to help.
